### PR TITLE
feat: issue #17 verification HTTP API for browser-triggered jobs

### PR DESCRIPTION
**Pull request by Th0rgal**
## Summary
Implements the missing browser-callable service layer for Issue #17 on top of the existing deterministic verification workflow core.

## What was implemented
- Added deterministic verification API server module:
  - `src/verification-api.ts`
  - `startVerificationHttpServer(...)`
  - `createChildProcessVerificationRunner(...)`
- Added HTTP routes for browser integration:
  - `GET /health`
  - `POST /api/verification/jobs`
  - `GET /api/verification/jobs`
  - `GET /api/verification/jobs/:jobId`
  - `POST /api/verification/jobs/:jobId/run`
  - `POST /api/verification/run-next`
- Added canonical hash surfacing (`jobHash`) for auditable UI/query responses.
- Added ledger-backed resume behavior (load-on-start, persist-on-enqueue/run transitions).
- Added server script:
  - `scripts/verification-server.mjs`
  - `npm run serve:verification`
- Added tests:
  - `tests/verification-api.test.ts`
  - Covers enqueue/list/run/get, reload-from-ledger, run-next when empty, invalid request and unknown routes.
- Updated exports/docs/readme:
  - `src/index.ts`
  - `docs/verification-api.md`
  - `docs/verification-flow.md`
  - `README.md`

## Determinism / provenance
- Canonical `VerificationWorkflow` records are returned directly from API routes.
- Every job response includes a deterministic SHA-256 job hash.
- Ledger persistence uses canonicalized workflow serialization.
- Invalid payloads return explicit machine-checkable API errors (`400 invalid_request`).

## Local validation
- `npm test` -> pass (`73/73`)
- `npm run build` -> pass
- `timeout 3 npm run serve:verification` -> starts server and prints deterministic URL/ledger path output

## What remains
1. Connect these endpoints to the frontend leaf panel and tree UI actions (`#15`, `#16`).
2. Add auth/multi-tenant policy around verification endpoints if needed for hosted deployments.
3. Add CI workflow job to run API integration tests in GitHub Actions once issue `#21` is implemented.

## How this advances objective
This removes the main integration gap between the provenance-preserving verification core and browser workflows: leaves can now trigger/query reproducible proof checks through deterministic, ledger-backed endpoints suitable for UI wiring.

Closes part of #17.

---

> [!NOTE]
> **Medium Risk**
> Introduces a new HTTP server adapter and a child-process runner that executes verification commands, which increases exposure to input validation and command-execution edge cases. Risk is mitigated by strict request canonicalization, deterministic responses, and integration tests covering persistence and error paths.
> 
> **Overview**
> Implements a new verification HTTP API (`startVerificationHttpServer`) on top of the existing `VerificationWorkflow`, including endpoints to **enqueue jobs**, **list/filter jobs**, **fetch a job with deterministic `jobHash`**, and **run jobs** (by id or `run-next`) while persisting every state transition to the verification ledger.
> 
> Adds a `createChildProcessVerificationRunner` (and `npm run serve:verification` script) to run reproducibility contracts via spawned commands with timeout handling, exports the new module from `src/index.ts`, and updates docs/README plus new tests that validate ledger-backed resume and deterministic error responses.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e943ce2f5413dd2cd2544498c36467f8f1f36f2d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by juanjose210879-dev — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Inductive explanation trees from Lean specifications (Verity -> Yul case study).
 - Issue #25 follow-up: prerequisite-order policy now evaluates deterministic grouping order (not lexical IDs) and waives cyclic in-group edges, fixing real Verity SCC compatibility.
 - Issue #25 follow-up: tree builder now reorders each group by local prerequisites and uses a safe depth guard with explicit no-progress failure diagnostics for large Lean corpora.
 - Issue #17: browser-triggered verification workflow core with deterministic queue/status lifecycle, reproducibility contracts, and canonical ledger persistence.
+- Issue #17 follow-up: browser-callable verification HTTP API with deterministic route payloads, persisted ledger-backed job querying, and command-runner integration.
 
 ## Local checks
```
Comment (juanjose210879-dev) [inline README.md]: Revisar si me estan hakeando
